### DISCUSSION TITLE
Full height table support?

### DISCUSSION THREAD

**Discussion by ckchiu2** [KevinVandy/material-react-table#56] [Q&A]
Is there any built-in support for rendering the table in 100% height of the parent component?

**Comment by KevinVandy** [comment]
Yes, here are a couple examples where you can either 

[Give a table container a min height](https://www.material-react-table.dev/?path=/story/styling-table-dimensions-examples--min-height)

```tsx
<MaterialReactTable
    columns={columns}
    data={data.slice(0, 5)}
    muiTableContainerProps={{
      sx: {
        minHeight: '800px',
      },
    }}
  />
```

or 

[Make the table expand to exactly the parent div's height](https://www.material-react-table.dev/?path=/story/styling-table-dimensions-examples--min-height-parent), though it takes a bit of calc of refs to get it right

```tsx
  <div style={{ height: '700px' }}>
    <MaterialReactTable
      columns={columns}
      data={data.slice(0, 5)}
      muiTableContainerProps={({ table }) => ({
        sx: {
          height: `calc(100% - ${table.refs.topToolbarRef.current?.offsetHeight}px - ${table.refs.bottomToolbarRef.current?.offsetHeight}px)`,
        },
      })}
      muiTablePaperProps={{
        sx: {
          height: '100%',
        },
      }}
    />
  </div>
```

**Reply by ckchiu2** [reply]
I was expecting something like 
```jsx
<div style={{ height: '700px' }}>
    <MaterialReactTable
      columns={columns}
      data={data.slice(0, 5)}
      muiTablePaperProps={{
        sx: {
          height: "100%",
          display: "flex",
          flexDirection: "column",
        }
      }}
      muiTableContainerProps={{
        sx: {
          flexGrow: 1,
          height: 0,
        }
      }}
    />
  </div>
```
Anyway, I think I have asked the wrong question. I want to apply the above solution using <ThemeProvider /> so that I do not need to mess up with the default props. However, I am lacking of a way to specify the only paper component that wraps the table. I think adding a class name for the table-wrapping paper at the library level could be a good solution. What do you think?

**Reply by KevinVandy** [reply]
I still not clear on what you want to accomplish?

**Reply by ckchiu2** [reply]
I want to use the following snippet to customize the css of the table paper without messing with the sx props every time I want a full height table. Without a classname I would also override the row action menu paper style.

```jsx
const theme = createTheme({
  components: {
    MuiPaper: {
      styleOverrides: {
        root: {
          "&.someclassname": {
            height: "100%",
            display: "flex",
            flexDirection: "column",
          }
        }
      }
    }
  }
})

<ThemeProvider theme={theme}>
  <MaterialReactTable 
  ...
  />
</ThemeProvider>
```

**Reply by zhongjiaguo001** [reply]
Followed your instructions, but it still won't work.
![image](https://github.com/KevinVandy/material-react-table/assets/44691489/db81a3b5-44ac-4ea1-a131-e6915a16f493)
![image](https://github.com/KevinVandy/material-react-table/assets/44691489/7250328c-3dec-4f3b-816f-34011a64d312)
![image](https://github.com/KevinVandy/material-react-table/assets/44691489/0d216a8a-5b53-475b-8eae-4b2f57470fb7)

**Reply by zhongjiaguo001** [reply]
Resolved, after troubleshooting it's a tier problem, thanks for the example!
